### PR TITLE
Limit virtctl builds during development to native architecture

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

Let bazel only build virtctl for the local native architecture to allow faster development cycles with frequent branch changes, and to improve the reactiveness by invocing less separate bazel calls while developing on the same branch.

In https://github.com/kubevirt/kubevirt/pull/9832 we started building quite a bit more virtctl architectures. Therfore the change.

CI and release jobs are not impacted by this change and will continue building everything.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by xpivarc — COMMENTED** [review]
/approve 
Makes perfect sense

**Comment by rmohr** [comment]
@fossedihelm I would also be ok to switch mine to `VIRTCTL_BUILD_ALL_ARCH` which you have in your PR. I have no preference.

**Review by fossedihelm — APPROVED** [review]
Thanks!

File: Makefile [hunk 1]
```diff
@@ -13,7 +13,7 @@ bazel-generate:
 	SYNC_VENDOR=true hack/dockerized "./hack/bazel-generate.sh"
 
 bazel-build:
-	hack/dockerized "export BUILD_ARCH=${BUILD_ARCH} && export DOCKER_TAG=${DOCKER_TAG} && hack/bazel-fmt.sh && ./hack/multi-arch.sh build"
+	hack/dockerized "export BUILD_ARCH=${BUILD_ARCH} && export DOCKER_TAG=${DOCKER_TAG} && export CI=${CI} && export KUBEVIRT_RELEASE=${KUBEVIRT_RELEASE} && hack/bazel-fmt.sh && ./hack/multi-arch.sh build"
```
Comment (zhlhahaha) [inline Makefile:16]: Roman, I am not quite understand why we need the environmental variable `CI` here.

Comment (xpivarc) [reply Makefile:16]: The `CI` is set in Prow jobs. This means if we are running the jobs we will build all architectures and will make sure they can be built.

Comment (zhlhahaha) [reply Makefile:16]: Thanks @xpivarc

**Pull request merged**